projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/deploy/DataJobsSynchronizer.java

```diff
@@ -68,35 +67,25 @@ public class DataJobsSynchronizer {
           "${datajobs.deployment.configuration.synchronization.task.initial.delay.ms:10000}")
   @SchedulerLock(name = "synchronizeDataJobsTask")
   public void synchronizeDataJobs() {
```

Comment:
synchronizeDataJobs is doing multiple things and getting pretty long. Consider breaking it into smaller methods that handle separate concerns separately.

Reply:
Agree, I tried to reduce it a bit in this PR.